### PR TITLE
Feature/pinterest v5 api support

### PR DESCRIPTION
**Pull request by fivetran-reneeli**
**Please provide your name and company**

**Link the issue/feature request which this PR is meant to address**

**Detail what changes this PR introduces and how this addresses the issue/feature request linked above.**

**How did you validate the changes introduced within this PR?**

**Which warehouse did you use to develop these changes?**

**Did you update the CHANGELOG?**

- [ ] Yes

**Did you update the dbt_project.yml files with the version upgrade (please leverage standard semantic versioning)? (In both your main project and integration_tests)**

- [ ] Yes

**Provide an emoji that best describes your current mood**
 
:dancer:

**Feedback**

We are so excited you decided to contribute to the Fivetran community dbt package! We continue to work to improve the packages and would greatly appreciate your [feedback](https://www.surveymonkey.com/r/DQ7K7WW) on our existing dbt packages or what you'd like to see next.

**PR Template** 
- [Community Pull Request Template](?expand=1&template=pull_request_template.md) (default)

- [Maintainer Pull Request Template](?expand=1&template=maintainer_pull_request_template.md) (to be used by maintainers)

### CODE REVIEW

**Comment by fivetran-reneeli** [comment]
holding off on docs until PR review is complete!

**Comment by fivetran-reneeli** [comment]
Btw, in the v5, `advertiser_id` has been replaced by `ad_account_id`. However, to keep our Pinterest Ads package standard with our other ad packages and to avoid another PR in ad reporting, I kept it as `advertiser_id`, renaming it from `ad_account_id` in the staging models.

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
Hey @fivetran-reneeli this is definitely a tricky update to apply when we do not have data to support the assumption of the API update. However, during my review it doesn't look like these changes particularly match the release notes we are seeing on our [release notes for Pinterest Ads](https://fivetran.com/docs/applications/pinterest-ads/changelog#july2023).

In particular some things I noticed that didn't match the release notes:
- In your changes you included the `ad_account_id` field to every model. However, per the release notes it seems that this is a new field **only** in the `ad_group_history` model. Was there a reason this was added to every model in place of the `advertiser_id`?
- Similarly, it seems that `ad_account_id` was not included in the `ad_group_history` model. Was there a reason for this omission?
- I noticed you added all the new fields to the staging models. Are these really needed? Our initial build of the package didn't include all the fields from the source, only the necessary ones we deemed relevant to the downstream models. If they do in fact make sense to include then I am open to adding them. But I worry that with our limited understanding of these new fields it may not make sense to include them.
- The field removals are definitely something we want to consider. It looks like you got all of those covered for the impacted fields in the advertiser_history table! ✅ 

Ultimately, I would like us to consider the above questions. Would you be able to take another pass at this PR and make any necessary updates to this model and the downstream ones? I would rather we not include all the new fields but instead put priority on removing the deprecated fields and adding any necessary new fields (like the ad_account_id for the ad_group_history).

**Comment by fivetran-reneeli** [comment]
In particular some things I noticed that didn't match the release notes:

> In your changes you included the ad_account_id field to every model. However, per the release notes it seems that this is a new field only in the ad_group_history model. Was there a reason this was added to every model in place of the advertiser_id?

So there was some confusion over this. According to the [pinterest v5 docs](https://developers.pinterest.com/docs/getting-started/migration/#Ads), it reads as advertiser_id is getting replaced by ad_account_id across several endpoints, including ads, ad groups, advertiser, and campaigns. But I suppose our connector is only swapping it out for  `ad_group_history` then?

image



> Similarly, it seems that ad_account_id was not included in the ad_group_history model. Was there a reason for this omission?

Ah, I overlooked this, will add it in. 

> I noticed you added all the new fields to the staging models. Are these really needed? Our initial build of the package didn't include all the fields from the source, only the necessary ones we deemed relevant to the downstream models. If they do in fact make sense to include then I am open to adding them. But I worry that with our limited understanding of these new fields it may not make sense to include them.

I don't believe I added all the new fields, just the ones I thought were relevant. For example, I chose not to include `bid_strategy_type` in ad_group_history. I didn't bring any fields into the transforms though, just removed deprecated ones.

But yeah, the new fields I added are just brought into staging, for the ones I thought may be relevant. I'm also ok with just not having them entirely given we don't have info on them.

**Comment by fivetran-joemarkiewicz** [comment]
Thanks for the detailed responses @fivetran-reneeli! 

I am comfortable with keeping the new fields in the staging model and comfortable with them not being included in the transform.

On the topic of the ad_account_id, I thought Calvin mentioned that they did not take this into account (outside of the ad group table) in order to make the migration easier. Can we confirm this with him? It would be a lot easier to just get data for this to fully validate... I may need to bite the bullet to serve an ad so we can get data 😞

**Comment by fivetran-reneeli** [comment]
> 
> On the topic of the ad_account_id, I thought Calvin mentioned that they did not take this into account (outside of the ad group table) in order to make the migration easier. Can we confirm this with him? It would be a lot easier to just get data for this to fully validate... I may need to bite the bullet to serve an ad so we can get data 😞

You know, actually I think he did say that, I just misunderstood. Let me follow up in our thread to confirm-- don't need to run an ad!

**Comment by fivetran-reneeli** [comment]
Hi @fivetran-joemarkiewicz , I switched back relevant models to use `advertiser_id`

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,17 @@
...
+
+| **Model** | **Removed**  | **New**   |
+|---|---|---|
+|  [stg_pinterest_ads__advertiser_history](https://fivetran.github.io/dbt_pinterest_source/#!/model/model.pinterest_source.stg_pinterest_ads__advertiser_history) | `billing_profile_status`, `billing_type`, `merchant_id`, `owner_user_id`, `status`  | `owner_username`, `permissions`  |
+|  [stg_pinterest_ads__campaign_history](https://fivetran.github.io/dbt_pinterest_source/#!/model/model.pinterest_source.stg_pinterest_ads__campaign_history) |`campaign_budget_optimization_enabled`  | `default_ad_group_budget_in_micro_currency`, `is_automated_campaign`, `is_campaign_budget_optimization`, `is_flexible_daily_budgets`  |
+|  [stg_pinterest_ads__ad_group_history](https://fivetran.github.io/dbt_pinterest_source/#!/model/model.pinterest_source.stg_pinterest_ads__ad_group_history) |  | `pacing_delivery_type`, `placement_group`, `summary_status`, `ad_account_id` |
+
+
+- In the v5 upgrade, `advertiser_id` has been replaced by `ad_account_id`. However, to keep our Pinterest Ads package standard with our other ad packages, we have kept it as `advertiser_id`.
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:13]: Is this still true? Maybe it would be worthwhile to include how we renamed `ad_account_id` for the pin promotion history and ad group models to be `advertiser_id` to be inline with the other naming conventions of the field across the schema/package.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,17 @@
+# dbt_pinterest_source v0.8.0
+[PR #25](https://github.com/fivetran/dbt_pinterest_source/pull/25) introduces the following changes:
+## Pinterest API v5 Updates
+- Following Pinterest Ads deprecating the v4 API on June 30, 2023 in place of v5, the Pinterest Ads Fivetran connector now leverages the Pinterest v5 API. The following fields have been deprecated/ introduced:
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:4]:
```suggestion
- Following Pinterest Ads deprecating the v4 API on June 30, 2023 in place of v5, the Pinterest Ads Fivetran connector now leverages the Pinterest v5 API. The following fields have been deprecated/introduced:
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,17 @@
...
+
+| **Model** | **Removed**  | **New**   |
+|---|---|---|
+|  [stg_pinterest_ads__advertiser_history](https://fivetran.github.io/dbt_pinterest_source/#!/model/model.pinterest_source.stg_pinterest_ads__advertiser_history) | `billing_profile_status`, `billing_type`, `merchant_id`, `owner_user_id`, `status`  | `owner_username`, `permissions`  |
+|  [stg_pinterest_ads__campaign_history](https://fivetran.github.io/dbt_pinterest_source/#!/model/model.pinterest_source.stg_pinterest_ads__campaign_history) |`campaign_budget_optimization_enabled`  | `default_ad_group_budget_in_micro_currency`, `is_automated_campaign`, `is_campaign_budget_optimization`, `is_flexible_daily_budgets`  |
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:9]: We never actually had `campaign_budget_optimization_enabled` in the package, correct? If so, we should probably omit it from the removed fields within the package.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,17 @@
+# dbt_pinterest_source v0.8.0
+[PR #25](https://github.com/fivetran/dbt_pinterest_source/pull/25) introduces the following changes:
+## Pinterest API v5 Updates
+- Following Pinterest Ads deprecating the v4 API on June 30, 2023 in place of v5, the Pinterest Ads Fivetran connector now leverages the Pinterest v5 API. The following fields have been deprecated/ introduced:
+
+| **Model** | **Removed**  | **New**   |
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:6]: It seems the `stg_pinterest_ads__pin_promotion_history` model is not included as one of the ones which experienced updates. Can you add that into the changelog?

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-reneeli thanks for making the requested updates! I do have a few final questions and suggestions to apply before this is all good to go. Let me know if you have questions!

File: macros/get_advertiser_history_columns.sql [hunk 5]
```diff
@@ -2,16 +2,13 @@
...
-    {"name": "billing_profile_status", "datatype": dbt.type_string()},
-    {"name": "billing_type", "datatype": dbt.type_string()},
     {"name": "country", "datatype": dbt.type_string()},
     {"name": "created_time", "datatype": dbt.type_timestamp()},
     {"name": "currency", "datatype": dbt.type_string()},
     {"name": "id", "datatype": dbt.type_string()},
-    {"name": "merchant_id", "datatype": dbt.type_string()},
+    {"name": "owner_username", "datatype": dbt.type_string()},
     {"name": "name", "datatype": dbt.type_string()},
-    {"name": "owner_user_id", "datatype": dbt.type_string()},
```
[the remark this answers is not in the document]
Comment (fivetran-reneeli) [reply macros/get_advertiser_history_columns.sql:13]: Ahh... I was going off the list in our ticket since the release notes weren't available when I made the edits and our PM confirmed, and owner_user_id was included in there. But now I see that it's not in the release notes as a deprecated field, so I'll add it back in

File: models/stg_pinterest_ads__advertiser_history.sql [hunk 6]
```diff
@@ -23,14 +23,11 @@ final as (
...
         id as advertiser_id,
         name as advertiser_name,
-        status as advertiser_status,
-        billing_profile_status,
-        billing_type,
         country,
         created_time as created_at,
         currency as currency_code,
-        merchant_id,
-        owner_user_id,
```
[the remark this answers is not in the document]
Comment (fivetran-reneeli) [reply models/stg_pinterest_ads__advertiser_history.sql:33]: thanks for catching this, added it back in!

File: CHANGELOG.md (hunk 1 above)
Comment (fivetran-reneeli) [reply CHANGELOG.md:13]: Good point, forgot to update this

File: CHANGELOG.md (hunk 3 above)
Comment (fivetran-reneeli) [reply CHANGELOG.md:9]: Good point, removed

File: CHANGELOG.md (hunk 4 above)
Comment (fivetran-reneeli) [reply CHANGELOG.md:6]: Added

**Comment by fivetran-reneeli** [comment]
Updated documentation as well as the PR template above.

File: CHANGELOG.md (hunk 1 above)
Comment (fivetran-joemarkiewicz) [reply CHANGELOG.md]: Small note, was `advertiser_id` replaced in the `ad_group_history` and `pin_promotion_history`? I was under the impression it didn't even exist previously and was just added as `ad_account_id`? If that is the case, can you reword this to be in line with this understanding.

**Review by fivetran-joemarkiewicz — COMMENTED** [review]
@fivetran-reneeli thanks for making the appropriate updates! This looks almost good to go. I only have one remaining note [here](https://github.com/fivetran/dbt_pinterest_source/pull/25#discussion_r1264014409). However, that is a fairly small note. Once that is updated this will be good to go. Therefore, I a comfortable approving this PR!

**Review by fivetran-joemarkiewicz — APPROVED** [review]

**Comment by fivetran-reneeli** [comment]
My bad, I didn't respond to your comment but I made the changelog is clearer! And thanks for catching the readme version-- updated
